data/data/azure/vnet/public-lb.tf

```diff
@@ -147,7 +148,7 @@ resource "azurerm_lb_outbound_rule" "public_lb_outbound_rule_v4" {
   count = var.use_ipv4 && var.azure_private && ! var.azure_outbound_user_defined_routing ? 1 : 0
 
   name                    = "outbound-rule-v4"
-  loadbalancer_id         = azurerm_lb.public.id
+  loadbalancer_id         = azurerm_lb.public[0].id
```

Comment:
Couldn't this resource be created when the load balancer doesn't exist (that is `var.azure_private == true && var.azure_outbound_user_defined_routing == true`) in which case I think we'd have an error?

Reply:
Fixed.